code/modules/power/singularity/emitter.dm

```diff
@@ -318,22 +335,25 @@
 /obj/machinery/power/emitter/screwdriver_act(mob/living/user, obj/item/item)
 	if(..())
 		return TRUE
+	if(contains_cascade_kit && !panel_open)
+		balloon_alert(user, "can't open!")
+		return TRUE
 	default_deconstruction_screwdriver(user, "[base_icon_state]_open", base_icon_state, item)
 	return TRUE
 
 /// Attempt to toggle the controls lock of the emitter
 /obj/machinery/power/emitter/proc/togglelock(mob/user)
 	if(obj_flags & EMAGGED)
-		to_chat(user, span_warning("The lock seems to be broken!"))
+		balloon_alert(user, "lock broken!")
 		return
 	if(!allowed(user))
-		to_chat(user, span_danger("Access denied."))
+		balloon_alert(user, "access denied")
 		return
 	if(!active)
-		to_chat(user, span_warning("The controls can only be locked when \the [src] is online!"))
+		balloon_alert(user, "turn off first!")
```

Comment:
checks if the emitter is active. if it's inactive, the check fails. should be "turn on first!" instead

Reply:
dont code at 2 am yall